dsc/workflows/base/base_workflow.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import annotations
+
+import logging
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Any, final
+
+from dsc.exceptions import InvalidDSpaceMetadataError
+from dsc.item_submission import ItemSubmission
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+logger = logging.getLogger(__name__)
+
+
+class BaseWorkflow(ABC):
+    """A base workflow class from which other workflow classes are derived."""
+
+    def __init__(
+        self,
+        workflow_name: str,
+        submission_system: str,
+        email_recipients: list[str],
+        metadata_mapping: dict,
+        s3_bucket: str,
+        s3_prefix: str | None,
+        collection_handle: str,
+        output_queue: str,
+    ) -> None:
+        """Initialize base instance.
+
+        Args:
+            workflow_name: The name of the workflow.
+            submission_system: The system to which item submissions will be sent (e.g.
+            DSpace@MIT)
```

Comment:
This reveals some ignorance about the greater context of the application, but when would it _not_ be `DSpace@MIT`?  For Dome?  

Would it make sense to make this only values from an `Enum`?  Is this used directly in the SQS message that then tells DSS something special?

Or... wonder if it would make sense for each workflow defined to define this on the class itself (i.e. a class level attribute, [here's an example from GeoHarvester](https://github.com/MITLibraries/geo-harvester/blob/main/harvester/records/sources/alma.py#L27))?  Instead of passing to the Workflow when instantiated.  Seems like any given workflow will only ever want to deposit into a single repository.

Reply:
If class level (aka workflow level) static attributes might make sense, do wonder about other things passed here.  For example: would `output_queue` also be pretty workflow specific?  or `collection_handle`?  I honestly don't know, but maybe worth considering.

Reply:
You're correct, currently everything is DSpace@MIT but eventually I would like DSS to point to Dome for DDC uploads so this is to accommodate that. And yes, defining the init params on the derived workflow classes makes a lot of sense since they would largely be static for the workflow. `s3_prefix` and `collection_handle` the only ones that could potentially be dynamic, though I'd like to lock that down with the content owners where possible.